internal/state/execution_test.go

```diff
@@ -241,44 +242,72 @@ func TestBeginBlockByzantineValidators(t *testing.T) {
 }
 
 func TestProcessProposal(t *testing.T) {
-	height := 1
-	runTest := func(txs types.Txs, expectAccept bool) {
-		ctx, cancel := context.WithCancel(context.Background())
-		defer cancel()
-
-		app := &testApp{}
-		cc := abciclient.NewLocalCreator(app)
-		logger := log.TestingLogger()
-		proxyApp := proxy.NewAppConns(cc, logger, proxy.NopMetrics())
-		err := proxyApp.Start(ctx)
-		require.NoError(t, err)
+	height := 2
+	txs := factory.MakeTenTxs(int64(height))
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
 
-		state, stateDB, _ := makeState(t, 1, height)
-		stateStore := sm.NewStore(stateDB)
-		blockStore := store.NewBlockStore(dbm.NewMemDB())
-
-		blockExec := sm.NewBlockExecutor(
-			stateStore,
-			logger,
-			proxyApp.Consensus(),
-			mmock.Mempool{},
-			sm.EmptyEvidencePool{},
-			blockStore,
-		)
+	app := abcimocks.NewBaseMock()
+	cc := abciclient.NewLocalCreator(app)
+	logger := log.TestingLogger()
+	proxyApp := proxy.NewAppConns(cc, logger, proxy.NopMetrics())
+	err := proxyApp.Start(ctx)
+	require.NoError(t, err)
+
+	state, stateDB, privVals := makeState(t, 1, height)
+	stateStore := sm.NewStore(stateDB)
+	blockStore := store.NewBlockStore(dbm.NewMemDB())
 
-		block, err := sf.MakeBlock(state, int64(height), new(types.Commit))
+	blockExec := sm.NewBlockExecutor(
+		stateStore,
+		logger,
+		proxyApp.Consensus(),
+		mmock.Mempool{},
+		sm.EmptyEvidencePool{},
+		blockStore,
+	)
+
+	block0, err := sf.MakeBlock(state, int64(height-1), new(types.Commit))
+	require.NoError(t, err)
+	lastCommitSig := []types.CommitSig{}
+	partSet, err := block0.MakePartSet(types.BlockPartSizeBytes)
+	require.NoError(t, err)
+	blockID := types.BlockID{Hash: block0.Hash(), PartSetHeader: partSet.Header()}
+	voteInfos := []abci.VoteInfo{}
+	for _, privVal := range privVals {
+		vote, err := factory.MakeVote(ctx, privVal, block0.Header.ChainID, 0, 0, 0, 2, blockID, time.Now())
+		require.NoError(t, err)
+		pk, err := privVal.GetPubKey(ctx)
 		require.NoError(t, err)
-		block.Txs = txs
-		acceptBlock, err := blockExec.ProcessProposal(ctx, block)
+		addr := pk.Address()
+		voteInfos = append(voteInfos, abci.VoteInfo{SignedLastBlock: true, Validator: abci.Validator{Address: addr, Power: 1000}})
```

Comment:
Optional for legibility:

```suggestion
		voteInfos = append(voteInfos, abci.VoteInfo{
			SignedLastBlock: true,
			Validator: abci.Validator{
				Address: addr, Power: 1000,
			},
		})
```